Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -51,10 +101,54 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
+            var newArray = ProcessArray(array, index);
+            return newArray;
+        }
+
+        private static int[] ProcessArray(int[] array, int index)
+        {
+            var isValid = IsValidCase(array, index);
+
+            if (!isValid)
+            {
+                return array;
+            }
+
+            array = Remove(array, index);
             return array;
         }
 
+        private static bool IsValidCase(int[] array, int index)
```

Comment:
Also very generic. Could be a better fit to have that also in RemoveAt. You can put this function name as comment above and it will serve the same purpose.

Reply:
done